### PR TITLE
MINOR need to get a new transformer for each get() call. can't share'em

### PR DESCRIPTION
**Pull request by norwood**
@jeyhunkarimov @mjsax @guozhangwang 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java [hunk 1]
```diff
@@ -38,6 +44,32 @@
     @Rule
     public final KStreamTestDriver driver = new KStreamTestDriver();
 
+    @Test
+    public void testToInternlValueTransformerSupplierSuppliesNewTransformers() {
+        ValueTransformerSupplier vts = createMock(ValueTransformerSupplier.class);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java]: nit: add `final`

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java [hunk 2]
```diff
@@ -38,6 +44,32 @@
...
+        expect(vts.get()).andReturn(null).times(3);
+        InternalValueTransformerWithKeySupplier ivtwks =
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java]: nit: add `final`

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java [hunk 3]
```diff
@@ -38,6 +44,32 @@
...
+        replay(vts);
+        ivtwks.get();
+        ivtwks.get();
+        ivtwks.get();
+        verify(vts);
+    }
+
+    @Test
+    public void testToInternalValueTransformerSupplierSuppliesNewTransformers() {
+        ValueTransformerWithKeySupplier vtwks = createMock(ValueTransformerWithKeySupplier.class);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java]: nit: add `final`

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java [hunk 4]
```diff
@@ -38,6 +44,32 @@
...
+        expect(vtwks.get()).andReturn(null).times(3);
+        InternalValueTransformerWithKeySupplier ivtwks =
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java]: nit: add `final`

**Review by mjsax — APPROVED** [review]
Just some nits.

LGTM.

**Comment by guozhangwang** [comment]
LGTM.

**Comment by jeyhunkarimov** [comment]
Sorry for the buggy leftover. Thanks @norwood for the fix.

**Pull request merged**